src/world/area_tik/tik_03/86F0F0.c

```diff
@@ -2,8 +2,49 @@
 
 #include "world/common/UnkPosFunc2.inc.c"
 
-INCLUDE_ASM(s32, "world/area_tik/tik_03/86F0F0", func_80240960_86F460);
+void partner_set_goal_pos(s32, s32);
+extern s32 D_80242EBC_8719BC;
+extern Npc* wPartnerNpc;
+extern s32 D_80242930_871430[4];
 
-INCLUDE_ASM(s32, "world/area_tik/tik_03/86F0F0", func_802409C0_86F4C0);
+ApiStatus func_80240960_86F460(Evt* script, s32 isInitialCall) {
+    PlayerStatus* player = &gPlayerStatus;
+    u32 i;
+    for(i = 0; i < ARRAY_COUNT(D_80242930_871430); i++) {
+        if (gCollisionStatus.currentFloor != D_80242930_871430[i]) {
+            continue;
+        }
+        if ((player->actionState == ACTION_STATE_GROUND_POUND) || (player->actionState == ACTION_STATE_ULTRA_POUND)) {
+            return ApiStatus_BLOCK;
+        }
+    }
+    return ApiStatus_DONE2;
+}
 
-INCLUDE_ASM(s32, "world/area_tik/tik_03/86F0F0", func_802409E0_86F4E0);
+ApiStatus func_802409C0_86F4C0(Evt* script, s32 isInitialCall) {
+    func_800EF300();
+    return ApiStatus_DONE2;
+}
+
+ApiStatus func_802409E0_86F4E0(Evt* script, s32 isInitialCall) {
+    PlayerStatus* player = &gPlayerStatus;
+    switch(D_80242EBC_8719BC){
```

Comment:
empty line before the switch, space after `switch`, space before opening bracket